src/main/java/emissary/id/TikaFilePlace.java

```diff
@@ -86,23 +83,46 @@ protected void configurePlace() throws IOException {
         }
     }
 
+    /**
+     * Iterates over multiple configured signature paths, and returns an array of input streams of the configured paths
+     *
+     * @return InputStream array
+     * @throws IOException if configured files does not exists
+     */
+    private InputStream[] getTikaSignatures() throws IOException {
+        List<InputStream> tikaSignatures = new ArrayList<>();
+
+        for (String tikaSignaturePath : tikaSignaturePaths) {
+            File mfile = new File(tikaSignaturePath);
+            if (!mfile.exists() || !mfile.canRead()) {
+                throw new IOException("Missing or unreadable TIKA_SIGNATURE_FILE " + tikaSignaturePath);
+            }
+
+            logger.debug("Tika Signature File:  " + tikaSignaturePath);
+            tikaSignatures.add(new FileInputStream(tikaSignaturePath));
+        }
+
+        return tikaSignatures.toArray(new InputStream[0]);
+    }
+
     /**
      * Use the Tika mime type (magic) detector to identify the file type
-     * 
+     *
      * @param d the IBaseDataObject payload to evaluate
      * @return mediaType
      */
     private MediaType detectType(IBaseDataObject d) throws Exception {
         Metadata metadata = new Metadata();
         InputStream input = TikaInputStream.get(d.data(), metadata);
+        metadata.set(Metadata.RESOURCE_NAME_KEY, d.getFilename());
```

Comment:
Does Tika use the filename in determining the type of a file in some cases? If so, it would be nice to be able to disable this behavior via emissary configuration. There are contexts where this behavior would not be desirable.

Reply:
Tika applies the filename as [a hint, if provided](https://github.com/apache/tika/blob/master/tika-core/src/main/java/org/apache/tika/mime/MimeTypes.java#L500): If the filename hint is the only MimeType found or it's equal to or a sub type of, an existing hint; it will be used. If not the filename hint will be ignored. 

If you still think it's necessary, I can add an option to disable the filename check in the Emissary configuration?